packages/core/contracts/zodiac/OptimisticOracleModule.sol

```diff
@@ -0,0 +1,232 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity ^0.8.6;
+
+import "@gnosis.pm/zodiac/contracts/core/Module.sol";
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+import "../oracle/implementation/Constants.sol";
+import "../oracle/interfaces/FinderInterface.sol";
+import "../oracle/interfaces/SkinnyOptimisticOracleInterface.sol";
+import "../oracle/interfaces/OracleAncillaryInterface.sol";
+import "../common/implementation/Lockable.sol";
+import "../common/interfaces/AddressWhitelistInterface.sol";
+import "../common/implementation/AncillaryData.sol";
+
+contract OptimisticOracleModule is Module, Lockable {
+    event OptimisticOracleModuleDeployed(address indexed owner, address indexed avatar, address target);
+
+    event TransactionsProposed(uint256 indexed proposalId, address indexed proposer, uint256 indexed proposalTime);
+
+    event TransactionExecuted(uint256 indexed proposalId, uint256 indexed transactionIndex);
+
+    event ProposalDeleted(uint256 indexed proposalId);
+
+    // Since finder is set during setUp, you will need to deploy a new Optimistic Oracle module if this address need to be changed in the future
+    FinderInterface public finder;
+
+    IERC20 public collateral;
+    uint64 public liveness;
+    // extra bond in addition to the final fee for the collateral type
+    uint256 public bond;
+    string public rules;
+    bytes32 public immutable identifier = "ZODIAC";
+    SkinnyOptimisticOracleInterface public skinnyOptimisticOracle;
+
+    struct Transaction {
+        address to;
+        uint256 value;
+        bytes data;
+        Enum.Operation operation;
+    }
+
+    struct Proposal {
+        Transaction[] transactions;
+        uint256 requestTime;
+        bytes ancillaryData;
+        bool status;
+    }
+
+    Proposal[] public proposals;
+
+    /**
```

Comment:
this natspec is done pretty inconsistently. If you are going to apply it to the constructor then you should do it on all the other functions, else get rid of it here.

Reply:
Comments are a WIP, will definitely improve this.